ext/ftp/php_ftp.c

```diff
@@ -1606,6 +1605,8 @@ PHP_FUNCTION(ftp_get_option)
 	zval		*z_ftp;
 	zend_long		option;
 	ftpbuf_t	*ftp;
+	char		returnbuffer[18];
+	unsigned char	*bytes = (unsigned char *)&ftp->externalip.s_addr;
```

Comment:
This doesn't look correct, as it uses &ftp before ftp is properly initialized (by zend_fetch_resource() on line 1615 below), since these are only used once (in the switch below), you can move their decls to that case line 1630-1631:

```
 		case PHP_FTP_OPT_EXTERNALIP: {
 			char returnbuffer[18];
 			unsigned char *bytes = (unsigned char *) &ftp->externalip.s_addr;
 			snprintf(returnbuffer, sizeof(returnbuffer), "%d.%d.%d.%d",bytes[0],bytes[1],bytes[2],bytes[3]);
 			RETURN_STRING(returnbuffer);
 			} break;
```